src/stats/mod.rs

```diff
@@ -422,3 +427,90 @@ impl PeerConnectionStats {
         }
     }
 }
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InboundRTPStats {
+    // RTCStats
+    #[serde(with = "serialize::instant_to_epoch_seconds")]
+    pub timestamp: Instant,
+    #[serde(rename = "type")]
+    pub stats_type: RTCStatsType,
+    pub id: String,
+
+    // RTCRtpStreamStats
+    pub ssrc: SSRC,
+    pub kind: &'static str, // Either "video" or "audio"
+    // TODO: Add tranportId
+    // TODO: Add codecId
+
+    // RTCReceivedRtpStreamStats
+    pub packets_received: u64,
+    // TODO: packetsLost
+    // TOOD: jitter(maybe, might be uattainable for the same reason as `framesDropped`)
+    // NB: `framesDropped` can't be produced since we aren't decoding, might be worth introducing a
+    // way for consumers to control this in the future.
+
+    // RTCInboundRtpStreamStats
+    pub track_identifier: String,
+    pub mid: String,
+    // TODO: `remoteId`
+    // NB: `framesDecoded`, `frameWidth`, frameHeight`, `framesPerSecond`, `qpSum`,
+    // `totalDecodeTime`, `totalInterFrameDelay`, and `totalSquaredInterFrameDelay` are all decoder
+    // specific values and can't be produced since we aren't decoding.
+    pub last_packet_received_timestamp: SystemTime,
+    pub header_bytes_received: u64,
+    // TODO: `packetsDiscarded`. This value only makes sense if we have jitter buffer, which we
+    // cannot assume.
+    // TODO: `fecPacketsReceived`, `fecPacketsDiscarded`
+    pub bytes_received: u64,
+    // TODO: `nackCount`
+    // TODO: `firCount`
+    // TODO:  `pliCount`
+    // NB: `totalProcessingDelay`, `estimatedPlayoutTimestamp`, `jitterBufferDelay`,
+    // `jitterBufferTargetDelay`, `jitterBufferEmittedCount`, `jitterBufferMinimumDelay`,
+    // `totalSamplesReceived`, `concealedSamples`, `silentConcealedSamples`, `concealmentEvents`,
+    // `insertedSamplesForDeceleration`, `removedSamplesForAcceleration`, `audioLevel`,
+    // `totalAudioEneregy`, `totalSampleDuration`, `framesReceived, and `decoderImplementation` are
+    // all decoder specific and can't be produced since we aren't decoding.
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OutboundRTPStats {
+    // RTCStats
+    #[serde(with = "serialize::instant_to_epoch_seconds")]
+    pub timestamp: Instant,
+    #[serde(rename = "type")]
+    pub stats_type: RTCStatsType,
+    pub id: String,
+
+    // RTCRtpStreamStats
+    pub ssrc: SSRC,
+    pub kind: &'static str, // Either "video" or "audio"
+    // TODO: Add tranportId
+    // TODO: Add codecId
+
+    // RTCSentRtpStreamStats
+    pub packets_sent: u64,
+    pub bytes_sent: u64,
+
+    // RTCOutboundRtpStreamStats
+    pub track_identifier: String, // NB: non-canon
```

Comment:
I don't quite understand why this isn't in the specification. One of these is emitted per simulcast SSRC in the cast of simulcast, but it would still be useful to include the track's identifier imo.

Reply:
I haven't had a whole lot of time to read through the specification, but my guess is that pending a formal specification draft, the different vendors all guessed at what would be useful for themselves.

Since Pion predates the draft recommendation, I'm not surprised that there are fields that aren't documented in any spec.

I was starting to wonder if this might be a place where webrtc-rs moves out in front of Pion in terms of coding to the spec, or if the long-term intention is to move in concert with Pion.

Reply:
This isn't actually ported from Pion, this code doesn't exist in Pion. FWIW I found the answer to why it's not in the specification, will push a commit that documents this.